include/hax_types.h

```diff
@@ -31,182 +31,52 @@
-    GUEST_DS_AR                                 = 0x0000481a,
-    GUEST_FS_AR                                 = 0x0000481c,
-    GUEST_GS_AR                                 = 0x0000481e,
-    GUEST_LDTR_AR                               = 0x00004820,
-    GUEST_TR_AR                                 = 0x00004822,
-
-    GUEST_ES_BASE                               = 0x00006806,
-    GUEST_CS_BASE                               = 0x00006808,
-    GUEST_SS_BASE                               = 0x0000680a,
-    GUEST_DS_BASE                               = 0x0000680c,
-    GUEST_FS_BASE                               = 0x0000680e,
-    GUEST_GS_BASE                               = 0x00006810,
-    GUEST_LDTR_BASE                             = 0x00006812,
-    GUEST_TR_BASE                               = 0x00006814,
-    GUEST_GDTR_BASE                             = 0x00006816,
-    GUEST_IDTR_BASE                             = 0x00006818,
-
-    GUEST_ES_LIMIT                              = 0x00004800,
-    GUEST_CS_LIMIT                              = 0x00004802,
-    GUEST_SS_LIMIT                              = 0x00004804,
-    GUEST_DS_LIMIT                              = 0x00004806,
-    GUEST_FS_LIMIT                              = 0x00004808,
-    GUEST_GS_LIMIT                              = 0x0000480a,
-    GUEST_LDTR_LIMIT                            = 0x0000480c,
-    GUEST_TR_LIMIT                              = 0x0000480e,
-    GUEST_GDTR_LIMIT                            = 0x00004810,
-    GUEST_IDTR_LIMIT                            = 0x00004812,
-
-    GUEST_VMCS_LINK_PTR                         = 0x00002800,
-    GUEST_DEBUGCTL                              = 0x00002802,
-    GUEST_PAT                                   = 0x00002804,
-    GUEST_EFER                                  = 0x00002806,
-    GUEST_PERF_GLOBAL_CTRL                      = 0x00002808,
-    GUEST_PDPTE0                                = 0x0000280a,
-    GUEST_PDPTE1                                = 0x0000280c,
-    GUEST_PDPTE2                                = 0x0000280e,
-    GUEST_PDPTE3                                = 0x00002810,
-
-    GUEST_DR7                                   = 0x0000681a,
-    GUEST_PENDING_DBE                           = 0x00006822,
-    GUEST_SYSENTER_CS                           = 0x0000482a,
-    GUEST_SYSENTER_ESP                          = 0x00006824,
-    GUEST_SYSENTER_EIP                          = 0x00006826,
-    GUEST_SMBASE                                = 0x00004828,
-    GUEST_INTERRUPTIBILITY                      = 0x00004824,
-    GUEST_ACTIVITY_STATE                        = 0x00004826,
-
-    /* Invalid encoding */
-    VMCS_NO_COMPONENT                           = 0x0000ffff
-};
-
-typedef enum component_index_t component_index_t;
+/* Detect architecture */
+// x86 (32-bit)
+#if defined(__i386__) || defined(_M_IX86)
+#define HAX_ARCH_X86_32
+#define ASMCALL __cdecl
+#endif
+// x86 (64-bit)
```

Comment:
nit: Just in case we run into a compiler that defines things differently, how about making this code more robust:

```
#if defined(__i386__) || defined(_M_IX86)
// 32-bit x86
#elif defined(__x86_64__) || defined(_M_X64)
// 64-bit x86
#else
#error "Unknown target CPU architecture"
#endif
```

Reply:
I see that you've done the same for the compiler and platform checks, thanks!